pkg/templates/templates_test.go

```diff
@@ -803,3 +856,81 @@ spec:
 	// Templates rock!
 	// Y2x1c3RlcjAwMDE=
 }
+
+func TestSetEncryptionConfig(t *testing.T) {
+	t.Parallel()
+	// Generate a 256 bit for AES-256. It can't be random so that the test results are deterministic.
+	keyBytesSize := 256 / 8
+	key := bytes.Repeat([]byte{byte('A')}, keyBytesSize)
+	otherKey := bytes.Repeat([]byte{byte('B')}, keyBytesSize)
+	iv := bytes.Repeat([]byte{byte('I')}, IVSize)
+	otherIV := bytes.Repeat([]byte{byte('I')}, IVSize)
+
+	tests := []struct {
+		encryptionConfig EncryptionConfig
+		expectedError    error
+	}{
+		{
+			EncryptionConfig{
+				EncryptionEnabled:    true,
+				AESKey:               key,
+				InitializationVector: iv,
+			}, nil,
+		},
+		{
+			EncryptionConfig{
+				DecryptionEnabled:    true,
+				AESKey:               otherKey,
+				InitializationVector: otherIV,
+			}, nil,
+		},
+		{
+			EncryptionConfig{
+				EncryptionEnabled: false,
+				DecryptionEnabled: false,
+			}, nil,
+		},
+		{
+			EncryptionConfig{
+				EncryptionEnabled: true,
+				AESKey:            []byte{byte('A')},
+			}, fmt.Errorf("%w: %s", ErrInvalidAESKey, "crypto/aes: invalid key size 1"),
+		},
+		{
+			EncryptionConfig{
+				EncryptionEnabled: true,
+				AESKey:            key,
+			}, ErrIVNotSet,
+		},
+		{
+			EncryptionConfig{
+				EncryptionEnabled:    true,
+				InitializationVector: []byte{byte('A')},
+			}, ErrAESKeyNotSet,
+		},
+		{
+			EncryptionConfig{
+				DecryptionEnabled:    true,
+				AESKey:               otherKey,
+				InitializationVector: []byte{byte('A')},
+			}, ErrInvalidIV,
+		},
+	}
+
+	var simpleClient kubernetes.Interface = fake.NewSimpleClientset()
+
+	config := Config{}
+	resolver, _ := NewResolver(&simpleClient, &rest.Config{}, config)
+
+	for _, test := range tests {
+		err := resolver.SetEncryptionConfig(test.encryptionConfig)
+
+		if err == nil || test.expectedError == nil {
+			if !(err == nil && test.expectedError == nil) {
+				t.Fatalf("expected error: %v, got: %v", test.expectedError, err)
+			}
+		} else if err.Error() != test.expectedError.Error() {
```

Comment:
Since we are using static errors that are sometimes wrapped, you should be able to to do `!errors.Is(err, test.expectedError)`.